src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1998,18 +2003,53 @@ public override IOperation VisitInvocation(IInvocationOperation operation, int?
             return new InvocationExpression(operation.TargetMethod, visitedInstance, operation.IsVirtual, visitedArguments, semanticModel: null, operation.Syntax, operation.Type, operation.ConstantValue, operation.IsImplicit);
         }
 
+        private int? _currentInitializerCaptureId = null;
+
         public override IOperation VisitObjectCreation(IObjectCreationOperation operation, int? captureIdForResult)
         {
             ImmutableArray<IArgumentOperation> visitedArgs = VisitArguments(operation.Arguments);
 
             // Initializer is removed from the tree and turned into a series of statements that assign to the created instance
-            var objectCreation = new ObjectCreationExpression(operation.Constructor, initializer: null, visitedArgs, semanticModel: null, operation.Syntax, operation.Type, operation.ConstantValue, operation.IsImplicit);
+            IOperation objectCreation = new ObjectCreationExpression(operation.Constructor, initializer: null, visitedArgs, semanticModel: null, operation.Syntax, operation.Type, operation.ConstantValue, operation.IsImplicit);
 
-            // PROTOTYPE(dataflow): For a non-null initializer, we'll need to assign the created object to a flow reference and rewrite all initializers to assign to/call functions on it, then return return the flow reference
+            // PROTOTYPE(dataflow): Handle collection initializers
+            if (operation.Initializer != null)
+            {
+                SpillEvalStack();
+
+                int? previousInitializerId = _currentInitializerCaptureId;
+
+                int currentInitializerCaptureId = _availableCaptureId++;
+                _currentInitializerCaptureId = currentInitializerCaptureId;
+
+                AddStatement(new FlowCapture(currentInitializerCaptureId, objectCreation.Syntax, objectCreation));
+
+                foreach (IOperation initializer in operation.Initializer.Initializers)
+                {
+                    AddStatement(Visit(initializer));
```

Comment:
>Visit(initializer) [](start = 33, length = 18)

It feel suspicious that we didn't have to add special handling for some forms of initializers. For example, how the following initializer would be handled?
```
var x = new C1 { C2 = { P1 = 3, P2 = 4 } };
```
given
```
class C1
{
    public C2 C2 { get; set; } = new C2();
}

class C2
{
    public C2()
    {
        System.Console.WriteLine("C2");
    }

    public int P1;
    public int P2;
}
```